### PR TITLE
[x86/Linux] implement TheUMEntryPrestub

### PR DESCRIPTION
**Pull request by seanshpark**
Initial code for x86 TheUMEntryPrestub

### CODE REVIEW

**Comment by seanshpark** [comment]
cc @parjong

File: src/vm/prestub.cpp [hunk 1]
```diff
@@ -1647,7 +1647,7 @@ PCODE MethodDesc::DoPrestub(MethodTable *pDispatchingMT)
 // use the prestub.
 //==========================================================================
 
-#ifdef _TARGET_X86_
+#if defined(_TARGET_X86_) && !defined(FEATURE_PAL)
 static PCODE g_UMThunkPreStub;
```
Comment (parjong) [inline src/vm/prestub.cpp]: How about using FEATURE_STUBS_AS_IL instead of FEATURE_PAL?

**Comment by parjong** [comment]
``GenerateUMThunkPrestub`` is also revised while bring up x86/Linux port. It would be better to roll back that changes as well.

**Comment by seanshpark** [comment]
@parjong, Good point, thank you.

**Comment by janvorli** [comment]
@seanshpark just to make sure I understand - so you want to move to FEATURE_STUBS_AS_IL on x86 Linux now?

**Comment by seanshpark** [comment]
> so you want to move to FEATURE_STUBS_AS_IL on x86 Linux now?

@janvori, actually idea is from @parjong having more experience than me.

**Comment by parjong** [comment]
@janvorli FEATURE_STUBS_AS_IL is already turned on for x86/Linux even though I'm internally working on x86/Linux port after disabling it. 

In my experience with arm/Linux port, FEATURE_STUBS_AS_IL seems to be necessary to enable proper unwinding, and thus I asked @seanshpark to work on it.

**Comment by janvorli** [comment]
@parjong I agree that it is a good thing to enable it. I have just thought that the plan after some initial going back and forth was to do the bringup with this disabled to match the way it works on Windows first and only after things work sufficiently (without GC and EH), we would start moving towards enabling the FEATURE_STUBS_AS_IL. 
But since you said that you had it locally disabled and you already got the hello world to print the message, it seems it is the right time to make it fully work.
So, I was mostly just curious.

**Comment by seanshpark** [comment]
Thank you :)

**Pull request merged**